03_Modules/Certificates/src/module/client/acme.ts

```diff
@@ -0,0 +1,223 @@
+// Copyright Contributors to the CitrineOS Project
+//
+// SPDX-License-Identifier: Apache 2.0
+
+import { IChargingStationCertificateAuthorityClient } from './interface';
+import { CacheNamespace, ICache, SystemConfig } from '@citrineos/base';
+import * as acme from 'acme-client';
+import { Client } from 'acme-client';
+import { ILogObj, Logger } from 'tslog';
+import fs from 'fs';
+import forge from 'node-forge';
+
+export class Acme implements IChargingStationCertificateAuthorityClient {
+  private readonly _directoryUrl: string = acme.directory.letsencrypt.staging;
+  private readonly _email: string | undefined;
+  // Key: serverId, Value: [cert chain, sub ca private key]
+  private _securityCertChainKeyMap: Map<string, [string, string]> = new Map();
+
+  private _client: Client | undefined;
+  private _cache: ICache;
+  private _logger: Logger<ILogObj>;
+
+  constructor(config: SystemConfig, cache: ICache, logger?: Logger<ILogObj>) {
+    this._logger = logger
+      ? logger.getSubLogger({ name: this.constructor.name })
+      : new Logger<ILogObj>({ name: this.constructor.name });
+
+    this._cache = cache;
+
+    config.util.networkConnection.websocketServers.forEach((server) => {
+      if (server.securityProfile === 3) {
+        try {
+          this._securityCertChainKeyMap.set(server.id, [
+            fs.readFileSync(
+              server.tlsCertificateChainFilePath as string,
+              'utf8',
+            ),
+            fs.readFileSync(
+              server.mtlsCertificateAuthorityKeyFilePath as string,
+              'utf8',
+            ),
+          ]);
+        } catch (error) {
+          this._logger.error(
+            'Unable to start Certificates module due to invalid security certificates for {}: {}',
+            server,
+            error,
+          );
+          throw error;
+        }
+      }
+    });
+
+    this._email = config.modules.certificates?.chargingStationCA.acme?.email;
+    const accountKey = fs.readFileSync(
+      config.modules.certificates?.chargingStationCA?.acme
+        ?.accountKeyFilePath as string,
+    );
+    const acmeEnv: string | undefined =
+      config.modules.certificates?.chargingStationCA?.acme?.env;
+    if (!acmeEnv && acmeEnv === 'production') {
```

Comment:
can this if statement be simplified to `acmeEnv === 'production'` ? just wondering if this condition works with that first condition `!acmeEnv` 🤔

Reply:
Thank you, Venus! You are right. I removed 1st condition.